server.ts

```diff
@@ -32,108 +32,138 @@ app.get("/resources", async (req, res) => {
+    if (isThereUser.rowCount < 1) {
+      res.status(400).send({ error: `No User in database matching that id (${userId})` })
+    } else {
+      const dbres = await client.query(`SELECT * FROM study_list WHERE author_id = $1`, [userId])
       res.status(200).json(dbres.rows);
     }
   } catch (error) {
-    res.status(500).send({error: error, stack: error.stack})
+    res.status(500).send({ error: error, stack: error.stack })
   }
 })
 
 //Get all tags
-app.get("/tags", async(req,res) => {
+app.get("/tags", async (req, res) => {
   try {
     const dbres = await client.query(`SELECT * FROM tags`)
     res.status(200).json(dbres.rows);
   } catch (error) {
-    res.status(500).send({error: error, stack: error.stack})
+    res.status(500).send({ error: error, stack: error.stack })
   }
 })
 
 //Get all resources for a single tag
-app.get<{id: string},{},{}>("/tags/:id", async(req,res) => {
+app.get<{ id: string }, {}, {}>("/tags/:id", async (req, res) => {
   try {
     const tagId = parseInt(req.params.id);
     const isThereTag = await client.query('SELECT * FROM tags WHERE id = $1', [tagId]);
-    if(isThereTag.rowCount<1){
-      res.status(400).send({error: `No tag in database matching that id (${tagId})`})
-    }else{
+    if (isThereTag.rowCount < 1) {
+      res.status(400).send({ error: `No tag in database matching that id (${tagId})` })
+    } else {
       const dbres = await client.query(`
-      SELECT * FROM tag_assignments
+      SELECT
+        tag_assignments.id AS assignment_id,
+        tag_assignments.tag_id,
+        tags.name as tag_name,
+        tag_assignments.resource_id,
+        resources.*
+      FROM tag_assignments
       JOIN resources ON tag_assignments.resource_id = resources.id
       JOIN tags ON tag_assignments.tag_id = tags.id
       WHERE tag_assignments.tag_id = $1
-    `,[tagId]);
-      if(dbres.rowCount<1){
-        res.status(400).send({error: `No resources with (${tagId})`})
-      }else{
+
+    `, [tagId]);
+      if (dbres.rowCount < 1) {
+        res.status(400).send({ error: `No resources with (${tagId})` })
+      } else {
         res.status(200).json(dbres.rows)
       }
     }
   } catch (error) {
-    res.status(500).send({error: error, stack: error.stack})
+    res.status(500).send({ error: error, stack: error.stack })
   }
 })
 
+//Get votes for a single resource
+app.get<{ id: string }, {}, {}>("/resources/:id/votes", async (req, res) => {
+  try {
+    const resource_id = parseInt(req.params.id)
+
+    const votesTrueRes = await client.query(`
+    SELECT *
+    FROM votes
+    WHERE resource_id = $1 AND is_upvote = true;`, [resource_id])
+
+    const votesFalseRes = await client.query(`
+    SELECT *
+    FROM votes
+    WHERE resource_id = $1 AND is_upvote = false;`, [resource_id])
+
+    const totalVote = votesTrueRes.rowCount - votesFalseRes.rowCount
+
+    res.status(200).json({ votes: totalVote });
```

Comment:
Can the JSON response contain both upvotes, downvotes, as well as totalVote please so that we can show all 3 numbers on react